src/table/components/head/HeadCellMenu.tsx

```diff
@@ -40,20 +40,24 @@ export default function HeadCellMenu({
   headerStyle,
   translator,
   sortFromMenu,
+  menuState,
 }: {
   headerStyle: GeneratedStyling;
   translator: ExtendedTranslator;
   sortFromMenu: (evt: React.MouseEvent, sortOrder: string) => void;
+  menuState: (value: boolean) => void;
 }) {
   const [open, setOpen] = useState(false);
```

Comment:
this state is not needed anymore. There should only be one single source of truth. Pass both the state and the setter. Basically we want to move the state one step up

Reply:
I think the current implementation is fair. because the direct parent of `<HeadCellMenu />` component is `<TableHeadWrapper />` which iterates through all columns inside of it's return section and does all the logic for rendering there. 

So, the scenario that Donya has mentioned (tried it before -> toggling all dropdown at once) is make sense now, because `<TableHeadWrapper />` [is the holder of all columns in table](https://github.com/qlik-oss/sn-table/blob/eac5ee9375a9a234136ab56254d68231f2153355/src/table/components/head/TableHeadWrapper.tsx#L45), and not one of them (to manage the dropdown of only one cell)

## Solution: 
**The problem that i see here is:** that we don't have a middle component of `<HeadCell />` that would sit in between `<TableHeadWrapper />` and `<HeadCellMenu />` => which would handle the state of toggle dropdown and any other scoped logic for it's own. 

So by that I could propose distinct 2 solution:
1. If we would have the middle component, the logic that Max has mentioned would be kept in there + would also result in reducing testing complexity as well
2. if we still want to move state up, we have to keep track of which component is active in order to tie the dropdown open state into it.

2nd solution is kinda easy fix for this but **1st one the the right way to do it IMO**

Reply:
First of all, see my other comment about scoping the fix for clicking the header when menu is open. I don't think we should fix it at all. blocking a click for something else when the menu is open, we don't do that anywhere else.

If we need a new component, that is fine. I like the principle of developing and structuring code to what you know you need. So I am not a fan of point nr 4. But if we need it for the listbox and the refactor makes sense, we should do it.

Reply:
Thanks @a-m-dev for breaking it through and your solutions. Max had the idea that we can skip passing the menu open/close state to the parent `TableHeadWrapper` to prevent getting all menu open but stopping extra head cell sorting by checking the existence of context menu in the DOM which we faced an issue there.
BTW, if you still think you want to see the whole issue in a clean PR here you have it: https://github.com/qlik-oss/sn-table/pull/765

Reply:
@haxxmaxx ..so for being sure I got it properly..  you believe that we should skip the extra sorting on head cell issue when the click-away(to close the context menu) occurs on the head cell ? and you thinks it's because we might break the same pattern in the other dropdowns ?

Reply:
you should be able to click the header to sort when the menu is open. So the verifail can be closed as works as designed